include/OpenSpaceToolkit/Astrodynamics/EventCondition/COECondition.hpp

```diff
@@ -0,0 +1,226 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_EventConditions_COECondition__
+#define __OpenSpaceToolkit_Astrodynamics_EventConditions_COECondition__
+
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+#include <OpenSpaceToolkit/Core/Types/String.hpp>
+
+#include <OpenSpaceToolkit/Mathematics/Objects/Vector.hpp>
+
+#include <OpenSpaceToolkit/Physics/Units/Derived.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Length.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/EventCondition.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace eventcondition
+{
+
+using ostk::core::types::Real;
+using ostk::core::types::String;
+
+using ostk::math::obj::VectorXd;
+using ostk::math::obj::Vector3d;
+
+using ostk::physics::units::Angle;
+using ostk::physics::units::Derived;
+using ostk::physics::units::Length;
+
+using ostk::astro::EventCondition;
+
+/// @brief                      A Classical Orbital Element based event condition
+
+class COECondition : public EventCondition
+{
+   public:
+    enum class Element
+    {
+        SemiMajorAxis,
+        Eccentricity,
+        Inclination,
+        ArgumentOfPeriapsis,
+        RightAngleOfAscendingNode,
```

Comment:
```suggestion
        RightAscensionOfAscendingNode,
```

Could also use `Raan` to match what we use in [here](https://github.com/open-space-collective/open-space-toolkit-astrodynamics/blob/main/include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Kepler/COE.hpp)

Reply:
Same on `Aop` actually (if we want to align on `COE.hpp`)